### PR TITLE
feat(core): Introduce getRequiredOne method for AccountCredentialsRepository

### PR DESCRIPTION
**Pull request by Aloren**
And replace all duplicated code with the newly introduced method.

### CODE REVIEW

**Comment by fieldju** [comment]
Seems like the build is failing on this PR: https://github.com/spinnaker/kayenta/pull/663/checks?check_run_id=419649536#step:5:3787

**Comment by Aloren** [comment]
@fieldju thank you. Will fix that shortly.

**Comment by Aloren** [comment]
Also has one more look at the implementation and decided to move `getRequiredOne` method to the `AccountCredentialsRepository` interface and make it `default`, so that some external integrations do not break.

File: kayenta-core/src/test/groovy/com/netflix/kayenta/security/MapBackedAccountCredentialsRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Netflix, Inc.
```
Comment (duftler) [inline kayenta-core/src/test/groovy/com/netflix/kayenta/security/MapBackedAccountCredentialsRepositoryTest.java]: s/Netflix/your company name

Comment (duftler) [inline kayenta-core/src/test/groovy/com/netflix/kayenta/security/MapBackedAccountCredentialsRepositoryTest.java]:
```suggestion
 * Copyright 2020 Playtika, Inc.
```

Comment (Aloren) [reply kayenta-core/src/test/groovy/com/netflix/kayenta/security/MapBackedAccountCredentialsRepositoryTest.java]: fixed. thanks

**Comment by Aloren** [comment]
oops. accidentally have overwritten default method with the previous changes, fixed that.

**Pull request merged**